File: algos/binary-search/binary-search.go
func search(nums []int, target int) int {
	// using sort package
	// i := sort.SearchInts(nums, target)

	// if (i < len(nums) && nums[i] == target) {
	//     return i
	// }
	// return -1

	lo := 0
	hi := len(nums) - 1

	for lo <= hi {
		mid := lo + (hi-lo)/2
		if nums[mid] == target {
			return mid
		} else if nums[mid] < target {
			lo = mid + 1
		} else {
			hi = mid - 1
		}
	}
	return -1
}

File: algos/binary-search/binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        """
        approach:
        guess in middle of list
        if guess is to small, make new list
        starting from middle to end
        if guess too large, make new list
        starting from start to middle
        """

        L, R = 0, len(nums) - 1

        while L <= R:
            M = L + ((R - L) // 2) # // shorthand to floor val
            if nums[M] > target:
                R = M - 1
            elif nums[M] < target:
                L = M + 1
            else:
                return M
        return -1
